src/FrequencyMapAgent.cpp

```diff
@@ -364,22 +364,33 @@ namespace geopm
     std::vector<std::pair<std::string, std::string> > FrequencyMapAgent::report_host(void) const
     {
         std::vector<std::pair<std::string, std::string> > result;
-        std::ostringstream oss;
-        oss << std::setprecision(M_PRECISION) << std::scientific;
-        for (const auto &region : m_hash_freq_map) {
-            oss << "\n    0x" << std::hex << std::setfill('0') << std::setw(16) << std::fixed;
-            oss << region.first;
-            oss << std::setfill('\0') << std::setw(0) << std::scientific;
-            oss << ": " << region.second;
-        }
-        for (const auto &region : m_default_freq_hash) {
-            oss << "\n    0x" << std::hex << std::setfill('0') << std::setw(16) << std::fixed;
-            oss << region;
-            oss << std::setfill('\0') << std::setw(0) << std::scientific;
-            oss << ": " << m_default_freq;
+
+        std::map<uint64_t, double> full_map(m_hash_freq_map);
+        std::for_each(m_default_freq_hash.begin(), m_default_freq_hash.end(), [&](uint64_t key)
+        {
+            full_map.insert({ key, m_default_freq });
+        });
```

Comment:
I'd avoid `std::for_each`.  This is much simpler:

```suggestion      
        for (const auto &region : m_default_freq_hash) {
            full_map.insert({region, m_default_freq});
        }
```